libs/protos/protos/steps/sp_steps_transform.proto

```diff
@@ -10,10 +10,53 @@ enum TransformType {
   TRANSFORM_TYPE_DELETE_FIELD = 2;
   TRANSFORM_TYPE_OBFUSCATE_VALUE = 3;
   TRANSFORM_TYPE_MASK_VALUE = 4;
+  TRANSFORM_TYPE_TRUNCATE_VALUE = 5;
+  // TODO: type for delete all keys except specified ones
 }
 
 message TransformStep {
-  string path = 1;
-  string value = 2; // Should this be bytes? ~DS
+  string path = 1 [deprecated=true];
+  string value = 2 [deprecated=true]; // Should this be bytes? ~DS
   TransformType type = 3;
+
+  oneof options {
```

Comment:
This is a really elegant approach - we can start applying this "options" pattern for a bunch of other stuff in the future. 👍